### PR TITLE
Tests for ShoppingCar and Cart Item.

### PR DESCRIPTION
**Pull request by sshestaka**
1. Added test cases for ShoppingCar and Cart Item entities (controller, service and repository).

### CODE REVIEW

File: src/test/java/mate/academy/onlinebookstore/repository/CartItemRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:database/shoppingcarts/remove-all-shopping-carts.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:database/books/remove-all-books.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
```
Comment (olekskov) [inline src/test/java/mate/academy/onlinebookstore/repository/CartItemRepositoryTest.java]: SQL Setup is 7 times longer than the test itself.
Combine scripts please

File: src/test/java/mate/academy/onlinebookstore/service/CartItemServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                .setQuantity(RED_BOOK_QUANTITY);
+    }
+
+    private CartItemDto getCartitemDto() {
+        return new CartItemDto()
+                .setId(getCartitem().getId())
+                .setBookId(getCartitem().getBook().getId())
+                .setBookTitle(getCartitem().getBook().getTitle())
+                .setQuantity(getCartitem().getQuantity());
+    }
```
Comment (olekskov) [inline src/test/java/mate/academy/onlinebookstore/service/CartItemServiceTest.java:114]: Please place private service methods after main logic

File: src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private BookService bookService;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+
+    private User getUserTest1() {
```
Comment (olekskov) [inline src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java]: same here

File: src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        CartItemDto cartItemDto = new CartItemDto();
+        cartItemDto.setId(cartItem.getId());
+        cartItemDto.setBookId(cartItem.getBook().getId());
+        cartItemDto.setBookTitle(cartItem.getBook().getTitle());
+        cartItemDto.setQuantity(cartItem.getQuantity() + 1);
+
+        ShoppingCartDto shoppingCartDto = new ShoppingCartDto()
+                .setId(shoppingCart.getId())
+                .setUserId(shoppingCart.getUser().getId())
+                .setCartItems(List.of(cartItemDto));
```
Comment (olekskov) [inline src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java]: Please extract these to separate methods.
Your test is too long

File: src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+        assertEquals(shoppingCartDtoActual, shoppingCartDto);
+    }
+
+    @Test
+    @DisplayName("Verify save() method add cart items works when given an existing cart item")
+    public void addCartItem_GivenCartItemWithNewBook_ShouldAddNewCartItem() {
+        User user = getUserTest1();
+        userRepository.save(user);
+
+        ShoppingCart shoppingCart = new ShoppingCart()
```
Comment (olekskov) [inline src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java]: same, do something, 70 lines of code for test method makes it harder to read than code itself

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 6]
```diff
@@ -258,4 +196,81 @@ void updateBook_UpdateExistedBook_Success() throws Exception {
...
+
+    private BookDto getBookDtoFromUpdatedBookRequestDto(UpdateBookRequestDto updateBookRequestDto) {
+        return new BookDto()
+                .setId(1L)
+                .setTitle(updateBookRequestDto.getTitle())
+                .setAuthor(updateBookRequestDto.getAuthor())
+                .setIsbn(updateBookRequestDto.getIsbn())
+                .setPrice(BigDecimal.valueOf(UPDATED_BOOK_PRICE))
+                .setDescription(updateBookRequestDto.getDescription())
+                .setCoverImage(updateBookRequestDto.getCoverImage());
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java:208]: You can create templates and reuse them for make more readability code

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 7]
```diff
@@ -258,4 +196,81 @@ void updateBook_UpdateExistedBook_Success() throws Exception {
...
+    }
+
+    private CreateBookRequestDto getTestBookDto() {
+        return new CreateBookRequestDto()
+                .setTitle("Test-Book")
+                .setAuthor("Test-Author")
+                .setIsbn("Test-ISBN")
+                .setPrice(BigDecimal.valueOf(TEST_BOOK_PRICE))
+                .setDescription("Test-Description")
+                .setCoverImage("Test-Cover-Image");
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java:274]: You can make it for all books

File: src/test/java/mate/academy/onlinebookstore/controller/CategoryControllerTest.java [hunk 8]
```diff
@@ -288,4 +255,40 @@ void getBooksByCategoryId_GivenCategoriesWithBooks_ShouldReturnAllBooksByCategor
...
+    private BookDtoWithoutCategoryIds getBlackBookDtoWithPrice21_99() {
+        return new BookDtoWithoutCategoryIds()
+                .setId(3L)
+                .setTitle("Black Book")
+                .setAuthor("Black Author")
+                .setIsbn("Black-ISBN")
+                .setPrice(BigDecimal.valueOf(21.99))
+                .setDescription("Black description")
+                .setCoverImage("Black cover image");
+
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/CategoryControllerTest.java:292]: Templates

File: src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+                .setCartItems(cartItemDto);
+    }
+
+    private CartItemDto getCartItemDto(CartItem cartItem) {
+        return new CartItemDto()
+                .setId(cartItem.getId())
+                .setBookId(cartItem.getBook().getId())
+                .setBookTitle(cartItem.getBook().getTitle())
+                .setQuantity(cartItem.getQuantity() + 1);
+    }
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/service/ShoppingCartServiceTest.java:308]: Make templates

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Pull request merged**